### PR TITLE
feat: Migrate Camera Management example to V3

### LINKED ISSUE #194: [Edgex 3.0] Upgrade App Service Examples (minnesota-dev branch) to V3
# Must do this on the on the minnesota-dev branch
Update the following custom app service examples .
- [x] simple-cbor-filter
- [x] simple-filter-xml
- [x] simple-filter-xml-http
- [x] simple-filter-xml-mqtt
- [x] advanced-filter-convert-publish
- [x] advanced-target-type
- [x] cloud-event
- [x] cloud-export-mqtt
- [x] custom-trigger
- [x] custom-trigger-nats-rpc
- [x] fledge-export
- [x] influxdb-export
- [x] json-logic
- [x] secrets
- [x] send-command
- [x] Camera Mgmt (waiting for Levski PR https://github.com/edgexfoundry/edgex-examples/pull/197 to merge)

And the following configurable examples:
- [x] app-service-configurable-ibm

Check them off as they are completed.

### PR DESCRIPTION
**Pull request by lenny-intel**
closes #194

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?) **N/A**
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)

## Testing Instructions
run non-secure EdgeX stack
run make build
run ./app-camera-management -cp -d -o
Verify service starts without error.

### CODE REVIEW

**Review by ajcasagrande — COMMENTED** [review]
Code looks good. Going to test it out.

**Comment by ajcasagrande** [comment]
We have a helper command in the makefile that is failing. Probably needs to be updated.

```
make run-app                                                                                                2 ✘ 
EDGEX_SECURITY_SECRET_STORE=false ./app-camera-management
level=INFO ts=2023-05-17T22:14:37.510039985Z app=app-camera-management source=service.go:490 msg="Starting app-camera-management 0.0.0 "
level=INFO ts=2023-05-17T22:14:37.510110514Z app=app-camera-management source=config.go:640 msg="Loading configuration file from res/configuration.yaml"
level=INFO ts=2023-05-17T22:14:37.510323194Z app=app-camera-management source=config.go:226 msg="Private configuration loaded from file with 0 overrides applied"
level=ERROR ts=2023-05-17T22:14:37.510677004Z app=app-camera-management source=messaging.go:45 msg="MessageBus configuration not set or missing from service's GetBootstrap() implementation"
level=ERROR ts=2023-05-17T22:14:37.510690416Z app=app-camera-management source=factory.go:43 msg="App Service initialization failed: boostrapping failed"
error: unable to create new app service app-camera-management!
make: *** [Makefile:29: run-app] Error 255
```

**Comment by ajcasagrande** [comment]
Error with the config of evam url, or parsing:

```
level=ERROR ts=2023-05-17T22:18:26.229103635Z app=app-camera-management source=app.go:56 msg="Unable to query EVAM pipeline statuses. Is EVAM running? GET request to query EVAM pipeline statuses failed: fail to parse baseUrl -> parse \"http://localhost:8080pipelines/status\": invalid port \":8080pipelines\" after host"
level=ERROR ts=2023-05-17T22:18:26.233954785Z app=app-camera-management source=app.go:61 msg="no devices found: no devices registered yet for the device services device-onvif-camera or device-usb-camera"
```

**Comment by ajcasagrande** [comment]
image

**Comment by lenny-intel** [comment]
> We have a helper command in the makefile that is failing. Probably needs to be updated.

Yep, needed the `-cp -d` flags. Added

**Comment by lenny-intel** [comment]
> Error with the config of evam url, or parsing:

I think the config value now need a trailing slash. Added it.

**Review by presatish — COMMENTED** [review]
Need to update the copyrights of updated files.

File: application-services/custom/camera-management/appcamera/evam.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func (app *CameraManagementApp) startPipeline(deviceName string, sr StartPipelin
 	if err != nil {
 		return err
 	}
-	reqPath := path.Join("pipelines", info.Name, info.Version)
+	reqPath := path.Join("/pipelines", info.Name, info.Version)
```
Comment (ajcasagrande) [inline application-services/custom/camera-management/appcamera/evam.go:136]: you have to fix in all of the calls.

**Comment by vyshali-chitikeshi** [comment]
Not able to run edge-analytics and saample app successfully, attached log
[edge_analytics_run_error.txt](https://github.com/edgexfoundry/edgex-examples/files/11503031/edge_analytics_run_error.txt)
[sample_app_run_error.txt](https://github.com/edgexfoundry/edgex-examples/files/11503032/sample_app_run_error.txt)

**Comment by ajcasagrande** [comment]
@lenny-intel looks like readme links and examples needs to be updated from toml to yaml

**Comment by lenny-intel** [comment]
> looks like readme links and examples needs to be updated from toml to yaml

Yep, dang embedded TOML. On it.

File: application-services/custom/camera-management/go.mod [hunk 2]
```diff
@@ -8,70 +8,70 @@ module github.com/edgexfoundry/edgex-examples/application-services/custom/camera
 go 1.18
 
 require (
-	github.com/IOTechSystems/onvif v0.0.2-0.20220301065030-7cf2dd734897
-	github.com/edgexfoundry/app-functions-sdk-go/v2 v2.3.0
-	github.com/edgexfoundry/go-mod-bootstrap/v2 v2.3.0
-	github.com/edgexfoundry/go-mod-core-contracts/v2 v2.3.0
+	github.com/IOTechSystems/onvif v0.1.6
+	github.com/edgexfoundry/app-functions-sdk-go/v3 v3.0.0-dev.66
```
Comment (lenny-intel) [inline application-services/custom/camera-management/go.mod:12]: @presatish , the SDK version can now be bumped to v3.0.0. Make tidy will bump the others

**Review by ajcasagrande — DISMISSED** [review]

**Review by ajcasagrande — APPROVED** [review]

**Pull request merged**